src/session/session_api.c

```diff
@@ -931,20 +923,28 @@ __session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
     WT_SESSION_IMPL *session;
 
     session = (WT_SESSION_IMPL *)wt_session;
+    SESSION_API_CALL(session, ret, alter, config, cfg);
+    /*
+     * We replace the default configuration listing with the current configuration. Otherwise the
+     * defaults for values that can be altered would override settings used by the user in create.
+     */
+    cfg[0] = cfg[1];
+    cfg[1] = NULL;
 
     /*
      * Alter table can return EBUSY error when the table is modified in parallel by eviction. Retry
      * the command after performing a system wide checkpoint. Only retry once to avoid potentially
      * waiting forever.
      */
-    ret = __session_alter_internal(session, uri, config);
+    ret = __session_alter_internal(session, uri, cfg);
```

Comment:
I'd recommend
```suggestion
    WT_ERR_ERROR_OK(__session_alter_internal(session, uri, cfg), EBUSY, true);
```
here so we explicitly jump to the error label on a non-`EBUSY` error. It's functionally identical to the current code but adds a bit of future proofing on the off chance we add new code later.

Reply:
Thank you Andrew!